### PR TITLE
bluetooth: doc: Misc fixes to Bluetooth GAP Doxygen documentation

### PR DESCRIPTION
**Pull request by kartben**
Various clean-ups, see individual commits.

### CODE REVIEW

File: include/zephyr/bluetooth/gap.h [hunk 1]
```diff
@@ -21,369 +21,693 @@ extern "C" {
...
+#define BT_COMP_ID_LF           0x05f1 /**< The Linux Foundation */
+/**
+ * @}
+ */
+
+/**
+ * @name EIR/AD data type definitions
+ * @{
+ */
+#define BT_DATA_FLAGS                   0x01 /**< AD flags */
```
Comment (Thalley) [inline include/zephyr/bluetooth/gap.h:41]: Does the `<` indicate that this comment applies to the value coming before it?

File: include/zephyr/bluetooth/gap.h [hunk 2]
```diff
@@ -21,369 +21,693 @@ extern "C" {
...
+ */
+
+/**
+ * @name Appearance Values
+ *
+ * Last Modified on 2023-01-05
+ * @{
+ */
+/** Generic Unknown */
+#define BT_APPEARANCE_UNKNOWN                               0x0000
```
Comment (Thalley) [inline include/zephyr/bluetooth/gap.h:102]: Above you kept the "documentation" for each value on the same line, but here you moved it to the previous line. Why the inconsistency?

File: include/zephyr/bluetooth/gap.h [hunk 3]
```diff
@@ -502,15 +839,15 @@ enum {
...
-	BT_GAP_SCA_UNKNOWN = 0,
-	BT_GAP_SCA_251_500 = 0,
-	BT_GAP_SCA_151_250 = 1,
-	BT_GAP_SCA_101_150 = 2,
-	BT_GAP_SCA_76_100 = 3,
-	BT_GAP_SCA_51_75 = 4,
-	BT_GAP_SCA_31_50 = 5,
-	BT_GAP_SCA_21_30 = 6,
-	BT_GAP_SCA_0_20 = 7,
+	BT_GAP_SCA_UNKNOWN = 0,   /**< Unknown */
```
Comment (Thalley) [inline include/zephyr/bluetooth/gap.h:842]: Ditto w.r.t. same line versus the line before it

**Review by Thalley — COMMENTED** [review]
Thanks for fixing something that has slightly annoyed me for ages :D 

A few questions

File: include/zephyr/bluetooth/gap.h (hunk 1 above)
Comment (kartben) [reply include/zephyr/bluetooth/gap.h:41]: Yep! https://www.doxygen.nl/manual/docblocks.html#memberdoc

File: include/zephyr/bluetooth/gap.h (hunk 3 above)
Comment (kartben) [reply include/zephyr/bluetooth/gap.h:842]: Happy to change everything to "line before" construct, if you think this is more consistent that way? I like how the `/**<` construct keeps things more concise though.

Comment (Thalley) [reply include/zephyr/bluetooth/gap.h:842]: We use `/**<` plenty of places in Zephyr that it's fine :D I was mostly curious as why you've opted for both ways in this PR :D

File: include/zephyr/bluetooth/gap.h (hunk 2 above)
Comment (kartben) [reply include/zephyr/bluetooth/gap.h:102]: Looks like I forgot to submit the most important reply/comment, before adding the others. So, the explanation is in commit #fff60f5. BT_APPEARANCE_* lines were already borderline too long, and definitely too long when adding the extra `*<`, so I add to split the comments in two lines for these.

**Review by Thalley — APPROVED** [review]

**Review by alwa-nordic — APPROVED** [review]

**Pull request merged**